master/buildbot/process/buildstep.py

```diff
@@ -1067,10 +1067,14 @@ def bad(arg):
             else:
                 setattr(self, arg, constructorArgs[arg])
             del constructorArgs[arg]
+        badargs = set()
         for arg in constructorArgs.keys():
```

Comment:
the fix is to use ```for arg in constructorArgs.keys()``` instead of ```for arg in constructorArgs:```

Difference is that keys returns a copy of the keys while constructorArgs return iterator